src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -34,6 +37,7 @@ public RemoteWalkContext(SourceCacheContext cacheContext, ILogger logger)
         public IList<IDependencyProvider> ProjectLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> LocalLibraryProviders { get; }
         public IList<IRemoteDependencyProvider> RemoteLibraryProviders { get; }
```

Comment:
Can we change access modifier to not public?
Not sure how many things it might affect, but that way we can prevent people accidentally using `RemoteLibraryProviders` which is not filtered out by `package namespace filter`.

Reply:
`RemoteWalkContext` instance has been created in another library i.e. `NuGet.Commands`. Hence it requires further refactoring to change that.

https://github.com/NuGet/NuGet.Client/blob/63b8dbad9399c4eaade9a2ccdecee7056a69bc7a/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L1164-L1184

Reply:
fixed in the latest commit.

Reply:
What was actual fix here? I still see `public` modifier.

Reply:
@erdembayar - Sorry for the confusion (copy/paste error)

Removing `public` access modifier for `RemoteDependencyProviders` has ripple effect in the API. I have couple of questions. For e.g. 

- Why in the existing code `RemoteLibraryProvider` were appended to a `new List` even though the source `request.DependencyProviders.RemoteProviders` is a readonly list?
https://github.com/NuGet/NuGet.Client/blob/5353034fed272cb81fbe60326f334b99871d8b74/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L1175-L1178

- What happens if I pass a reference to `request.DependencyProviders.RemoteProviders (IReadOnlyList)` directly to the `RemoteWalkContext`? Are there is other problems I am missing due to lack of total context.

- Should I make the `LocalLibraryProviders` also private? I decided that this is out of scope because this PR filter only `RemoteLibraryProvider`. If needed, we can do this in a quality week.


Given above information, I tried to address this comment in the latest commit https://github.com/NuGet/NuGet.Client/pull/4149/commits/a169d21f8ec4ed33259849f2892685c01d95bb9c

Please review and let me know the feedback. 

@nkolev92 - please take a quick look at the latest commit and let me know the feedback.

Reply:
I would keep it simple and keep the existing style. 

Add a public prop, with only a getter.

Reply:
My concern was having public modifier might let someone to use it without namespace filter by accident.

Reply:
reverted a169d21 commit from this PR as per offline conversation.